models/terminator/materials/scripts/ow_terrain.frag

```diff
@@ -226,14 +225,12 @@ void lighting(vec3 wsDirToSun, vec3 wsDirToEye, vec3 wsNormal, vec4 wsDetailNorm
   // Phong specular model we are currently using.
   vec3 vsDirToEye = normalize(normalMatrix * wsDirToEye);
   vec3 vsDetailNormal = normalize(normalMatrix * wsDetailNormalHeight.xyz);
-  spotlight(vsSpotlightPos0.xyz - vsPos, -vsSpotlightDir0.xyz, spotlightAtten0,
-            spotlightParams0.xyz, spotlightColor0.rgb * spotlightIntensityScale0,
-            spotlightTexCoord0, vsDirToEye, vsDetailNormal, specular_power,
-            diffuse, specular);
-  spotlight(vsSpotlightPos1.xyz - vsPos, -vsSpotlightDir1.xyz, spotlightAtten0,
-            spotlightParams0.xyz, spotlightColor0.rgb * spotlightIntensityScale1,
-            spotlightTexCoord1, vsDirToEye, vsDetailNormal, specular_power,
-            diffuse, specular);
+  for (int i=0; i<2; i++) {
+    spotlight(vsLightPos[i+1].xyz - vsPos, -vsLightDir[i+1].xyz, spotlightAtten0,
+              spotlightParams[1], spotlightColor0.rgb * spotlightIntensityScale[i],
```

Comment:
Does this `spotlightParams[1]` need to be `spotlightParams[i or i+1]` instead?

Reply:
Good catch! Fixed.